typescript/005-resistor-color-trio/resistor-color-trio.ts

```diff
@@ -32,18 +32,14 @@ export function decodedResistorValue([color1, color2, color3]: Colors[]): string
 		exp += 1;
 	}
 
-	let finalExp = exp;
-	let finalSuffix = "ohms";
-
-	for (const [key, value] of Object.entries(SI_PREFIXES)) {
-		if (exp >= Number(key)) {
-			finalExp = exp - Number(key);
-			finalSuffix = `${value}ohms`;
+	for (const [key, value] of SI_PREFIXES) {
+		if (exp >= key) {
+			const finalExp: number = exp - key;
+			const multiplier: number = 10 ** finalExp;
+			const result = rawVal * multiplier;
+			return `${result} ${value}`;
 		}
 	}
 
-	const multiplier: number = 10 ** finalExp;
-	const result = rawVal * multiplier;
-
-	return `${result} ${finalSuffix}`;
+	return `${rawVal * 10 ** exp} ohms`;
```

Comment:
This is still a special case that can be avoided.
(Notice that you're doing `10 ** X` both here and in the loop, this is a marker for code duplication)